tests/test_util/test_util_typing.py

```diff
@@ -184,6 +187,11 @@ def test_restify_type_hints_containers():
                                            "[:py:obj:`None`]")
 
 
+def test_restify_Annotated():
+    assert restify(Annotated[str, "foo", "bar"]) == ':py:class:`~typing.Annotated`\\ [:py:class:`str`]'
+    assert restify(Annotated[str, "foo", "bar"], 'smart') == ':py:class:`~typing.Annotated`\\ [:py:class:`str`]'
```

Comment:
As above, it'd be really valuable for users of e.g. [annotated-types](https://pypi.org/project/annotated-types/) to show the repr of the annotations too.

Reply:
I haven't used annotated types, could you provide a sketch of what would currently be output and what you'd prefer? Just so we can capture that in a test.

Reply:
```python
from typing import Annotated
from annotated_types import Gt, Lt, Len

def coinflip(
    p_heads: Annotated[float, Gt(0.0), Lt(1.0)],
    label: Annotated[str, Len(1, 10)] | None = None,
) -> bool:
    ...
```

would ideally render the signature as something like
> coinflip(p_heads: [Annotated](https://docs.python.org/3/library/typing.html#typing.Annotated)[[float](https://docs.python.org/3/library/functions.html#float), Gt(gt=0.0), Lt(lt=1.0)], label: [Annotated](https://docs.python.org/3/library/typing.html#typing.Annotated)[[str](https://docs.python.org/3/library/functions.html#func-str), Gt(gt=0.0), Lt(lt=1.0)]) -> [bool](https://docs.python.org/3/library/functions.html#bool)

using the `repr` of the dataclasses.